File: content/atlas/source/includes/fts/field-types/objectId/create-index-basic-placeholders.js
const { MongoClient } = require('mongodb');

// Connection URI
const uri = 'mongodb+srv://<username>:<password>@<cluster-url>';

// Create a new MongoClient
const client = new MongoClient(uri);

async function createSearchIndex() {
  try {
    await client.connect();
    const database = client.db('<databaseName>');

    // Create a MongoDB Search index on the objectId field
    await database.command({
      createSearchIndexes: '<collectionName>',
      indexes: [{
        name: '<indexName>',
        definition: {
          mappings: {
            dynamic: true | false,
            fields: {
              '<fieldName>': {
                type: 'objectId'
              }
            }
          }
        }
      }]
    });
    
    console.log('Successfully created Atlas Search index');
  } finally {
    await client.close();
  }
}

createSearchIndex().catch(console.error);

File: content/atlas/source/includes/fts/field-types/objectId/create_index_basic_placeholders.py
from pymongo import MongoClient

# Connect to your Atlas cluster
client = MongoClient("mongodb+srv://<username>:<password>@<cluster-url>")
db = client.sample_mflix

# Create a MongoDB Search index on the objectId field
r = db.command({
    "createSearchIndexes": "<collectionName>",
    "indexes": [{
        "name": "<indexName>",
        "definition": {
            "mappings": {
                "dynamic": True | False,
                "fields": {
                    "<fieldName>": {
                        "type": "objectId"
                    }
                }
            }
        }
    }]
})

print(r)
